internal/storage/storage.go

```diff
@@ -18,13 +18,18 @@ const (
 // EvaluationRule represents a rule and constraints required for evaluating if a
 // given flagKey matches a segment
 type EvaluationRule struct {
-	ID               string                 `json:"id"`
-	NamespaceKey     string                 `json:"namespace_key,omitempty"`
-	FlagKey          string                 `json:"flag_key,omitempty"`
-	SegmentKey       string                 `json:"segment_key,omitempty"`
-	SegmentMatchType flipt.MatchType        `json:"segment_match_type,omitempty"`
-	Rank             int32                  `json:"rank,omitempty"`
-	Constraints      []EvaluationConstraint `json:"constraints,omitempty"`
+	ID              string                        `json:"id"`
+	NamespaceKey    string                        `json:"namespace_key,omitempty"`
+	FlagKey         string                        `json:"flag_key,omitempty"`
+	Segments        map[string]*EvaluationSegment `json:"segments,omitempty"`
+	Rank            int32                         `json:"rank,omitempty"`
+	SegmentOperator flipt.SegmentOperator         `json:"segmentOperator,omitempty"`
+}
+
+type EvaluationSegment struct {
```

Comment:
we should probably put the JSON struct tags on this just to be safe since this will also be cached

Reply:
@markphelps Thank you for this. I could have just committed directly to this now that I think about it. Here is the PR: https://github.com/flipt-io/flipt/pull/1984